### ISSUE TITLE
torch.logcumsumexp loses a first-order JVP contribution

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#196705]
### 🐛 Describe the bug

`torch.logcumsumexp` returns `-5.9999999999994005` for a first-order JVP whose mathematically correct value is `-5.0` in `float64`.
Every prefix is a smooth log-sum-exp of finite logits. The one-element prefix must keep the derivative of its input. At `t = -400.0`, summing the explicit prefix log-sum-exp expressions gives JVP `-5.0`. PyTorch returns about `-6.0`, as if that length-1 contribution were dropped. The forward value matches.
This issue reproduces on CPU, so CUDA availability does not affect the result.
```python
import torch
dtype = torch.float64
def f(t):
    x = torch.stack((t, 1 - t, 2 * t + 3, -t, t + 2, 3 * t - 1)).reshape(2, 3)
    w = torch.tensor([[1, 2, -1], [3, -2, 4]], dtype=t.dtype)
    return (torch.logcumsumexp(x, dim=-1) * w).sum()
t = torch.tensor(-400.0, dtype=dtype)
_, actual = torch.func.jvp(f, (t,), (torch.ones_like(t),))
expected = torch.tensor(-5.0, dtype=dtype)
print("actual:")
print(actual)
print("expected:")
print(expected)
```
### Actual vs expected result
```text
actual:
-5.9999999999994005
expected:
-5.0
```
The discrepancy occurs in the first-order forward-mode derivative. The forward value is `2001.0` and matches the prefix log-sum-exp; the JVP is off by about `1`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A
OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64
Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01
CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128
Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @albanD

**Comment by indrajeetapache** [comment]
Working on a fix for this.

**Why this breaks:** `logcumsumexp_jvp` reused `logsumexp_jvp`'s single max-along-dim shift. For a prefix that sits far below a later maximum (e.g. first element `-400` with a later `401`), that prefix underflows to 0 in the normalized cumsum, so its JVP contribution becomes 0 instead of the input tangent. That drops the length-1 term and shifts the weighted JVP by about `1` (`-6` vs `-5`).

**Fix:** compute the JVP with the stable recurrence on the forward `logcumsumexp` values:
`y_t[i] = exp(y[i-1]-y[i])*y_t[i-1] + exp(x[i]-y[i])*x_t[i]`.

PR incoming.
